ext/canvas/canvas.rs

```diff
@@ -0,0 +1,249 @@
+use deno_core::cppgc::SameObject;
+use deno_core::op2;
+use deno_core::v8;
+use deno_core::webidl::UnrestrictedDouble;
+use deno_core::GarbageCollected;
+use deno_core::OpState;
+use deno_core::WebIDL;
+use deno_error::JsErrorBox;
+use image::ColorType;
+use image::DynamicImage;
+use image::GenericImageView;
+use std::cell::OnceCell;
+use std::cell::RefCell;
+use std::collections::HashMap;
+use std::rc::Rc;
+
+use crate::op_create_image_bitmap::ImageBitmap;
+
+struct BlobHandle(v8::Global<v8::Function>);
+
+pub type CreateCanvasContext = for<'s> fn(
+  canvas: v8::Global<v8::Object>,
+  data: Rc<RefCell<DynamicImage>>,
+  scope: &mut v8::HandleScope<'s>,
+  options: v8::Local<'s, v8::Value>,
+  prefix: &'static str,
+  context: &'static str,
+) -> Box<dyn CanvasContext>;
+
+pub struct RegisteredContexts(pub HashMap<String, CreateCanvasContext>);
+
+pub struct OffscreenCanvas {
+  data: Rc<RefCell<DynamicImage>>,
+
+  active_context: OnceCell<(String, Box<dyn CanvasContext>)>,
+}
```

Comment:
So to get hold of the V8 value, you could store contexts are V8 globals and unwrap them to `Box<dyn CanvasContext>` on usage; so you won't need `value()` method on `CanvasContext` trait.

```suggestion
pub type CreateCanvasContext = for<'s> fn(
  canvas: v8::Global<v8::Object>,
  data: Rc<RefCell<DynamicImage>>,
  scope: &mut v8::HandleScope<'s>,
  options: v8::Local<'s, v8::Value>,
  prefix: &'static str,
  context: &'static str,
) -> Global<Value>;

pub struct RegisteredContexts(pub HashMap<String, CreateCanvasContext>);

pub struct OffscreenCanvas {
  data: Rc<RefCell<DynamicImage>>,

  active_context: OnceCell<(String, Global<Value>)>,
}
```

and in the creation functions use `cppgc_make_object` to wrap CanvasContext in a V8 object.